### PR TITLE
[show][barefoot] replace shell=True

### PR DESCRIPTION
**Pull request by maipbui**
Signed-off-by: maipbui <maibui@microsoft.com>

#### What I did
`subprocess()` - when using with `shell=True` is dangerous. Using subprocess function without a static string can lead to command injection.
#### How I did it
`subprocess()` - use `shell=False` instead, use list of strings Ref: [https://semgrep.dev/docs/cheat-sheets/python-command-injection/#mitigation](https://semgrep.dev/docs/cheat-sheets/python-command-injection/#mitigation)

### CODE REVIEW

File: show/plugins/barefoot.py [hunk 1]
```diff
@@ -37,9 +39,10 @@ def profile():
     
     # Print profile list
     click.echo('Available profile(s):')
-    subprocess.run('docker exec -it syncd find /opt/bfn -mindepth 1 '
-        r'-maxdepth 1 -type d -name install_\*_profile ' + opts + '| sed '
-        r's%/opt/bfn/install_\\\(.\*\\\)_profile%\\1%', shell=True)
+    cmd0 = ['docker', 'exec', '-it', 'syncd', 'find', '/opt/bfn', '-mindepth', '1',\
+            r'-maxdepth', '1', r'-type', 'd', r'-name', r'install_\*_profile', opts]
```
Comment (qiluo-msft) [inline show/plugins/barefoot.py:43]: >r [](http://example.com/codeflow?start=12&length=1)

This `r` is not needed.

Comment (qiluo-msft) [inline show/plugins/barefoot.py:43]: >r [](http://example.com/codeflow?start=31&length=1)

the same

Comment (qiluo-msft) [inline show/plugins/barefoot.py:43]: >r [](http://example.com/codeflow?start=46&length=1)

the same

**Comment by qiluo-msft** [comment]
@oleksandrx-kolomeiets Could you help review?

**Review by qiluo-msft — APPROVED** [review]

**Comment by maipbui** [comment]
@oleksandrx-kolomeiets could you help review?

**Review by qiluo-msft — APPROVED** [review]

**Pull request merged**